frontend/src/views/main/project/TextExplanationParagraph.vue

```diff
@@ -0,0 +1,126 @@
+                min="0"
+                max="100"
+                label="Importance"
+            ></v-slider>
+        </div>
+        <VirtualCollection
+            :cellSizeAndPositionGetter="cellSizeAndPositionGetter"
+            :collection="items"
+            :height="200"
+            :width="widthParagraph"
+        >
+            <span slot="cell" slot-scope="props"  :style="{ backgroundColor: props.data.color }">{{props.data.text}}</span>
+        </VirtualCollection>
+    </div>
+</template>
+
+
+
+<script setup lang="ts">
+import { ref, getCurrentInstance, watch, onMounted } from 'vue';
+import VirtualCollection from "vue-virtual-collection/src/VirtualCollection.vue";
+
+const props = defineProps<{
+    weights: number[]
+    words: string[]
+    max_weight: number
+    min_weight: number
+}>();
+
+const paragraph = ref(null)
+const range = ref<number>(0)
+const widthParagraph = ref<number>(getCurrentInstance()?.proxy.$parent?.$el.clientWidth - 40);
+const x = ref<number>(0)
+const y = ref<number>(0)
+const classificationMinWeight = ref<number>(Math.min(...props.weights));
+const classificationMaxWeight = ref<number>(Math.max(...props.weights));
+const absoluteMaxWeight = ref<number>(Math.max(Math.abs(props.max_weight), Math.abs(props.min_weight)))
+const items =  ref<object[]>(createItems())
+
+// Changement of Classification Label
+watch(() => [...props.weights], () => {
+    classificationMinWeight.value = Math.min(...props.weights)
+    if (classificationMinWeight.value === 0)
+        classificationMinWeight.value = -1;
+    classificationMaxWeight.value = Math.max(...props.weights)
+    if (classificationMaxWeight.value === 0)
+        classificationMaxWeight.value = 1;
+})
+
+// Changement of Classification Label or Modification of the slider
+watch(() => [...props.weights, range.value], updateTextColor)
+
+
+function updateTextColor(){
+    for (let i = 0; i < props.words.length; i++){
+        items.value[i].data.color = calculateBackgroundColor(props.weights[i])  
+    } 
+}
+
+function calculateBackgroundColor(weight: number){
+    let multiplicatior = (weight < 0) ? weight / classificationMinWeight.value : weight / classificationMaxWeight.value;
+    multiplicatior *= 100;
+    if (multiplicatior < range.value || absoluteMaxWeight.value === 0){
+        return 'transparent';
+    }
+    else {
+        let color = (weight > 0) ? 'rgba(0, 256, 0, ' : 'rgba(256, 0, 0, ';
+        color += Math.abs(weight) / absoluteMaxWeight.value + ')'; 
+        return color
+    }
+}
+
+
+onMounted(() => {
+    window.addEventListener('resize', () => {
+        x.value = 0;
+        y.value = 0;
+        widthParagraph.value = paragraph.value.clientWidth; 
+        items.value = createItems()
+    });
+})
+
+// Called when mounted and when window resized 
+function createItems(){
+    return Array.from({length: props.words.length}, (_, index) => {
+        let c = document.createElement("canvas") as HTMLCanvasElement;
```

Comment:
you can create a single global canvas and use it for all words instead of creating a new one for each word can't you? In this case you won't even need to pass it as an argument to `getTextWidth` as it can be taken from component level.

Reply:
The problem is that the createItems function is called before every element is mounted on the page so we can't access a canvas for example. I modified so that the canvas is created only once and not for every element but it has to stay in the createItems function.